core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java

```diff
@@ -51,22 +52,30 @@
                 return extensionType + "@" + Integer.toHexString(System.identityHashCode(proxy));
             }
 
-            try {
-                if (threadExtension.get() != null) {
-                    return method.invoke(threadExtension.get(), args);
-                }
-                return db.withExtension(extensionType, extension ->
-                        JdbiThreadLocals.invokeInContext(threadExtension, extension,
-                                () -> method.invoke(extension, args)));
-            }
-            catch (InvocationTargetException e) {
-                throw e.getTargetException();
+            if (threadExtension.get() != null) {
+                return invoke(method, threadExtension.get(), args);
             }
+            return db.withExtension(extensionType, extension ->
+                    JdbiThreadLocals.invokeInContext(threadExtension, extension,
+                            () -> invoke(method, extension, args)));
         };
 
         return extensionType.cast(
                 Proxy.newProxyInstance(
                         extensionType.getClassLoader(),
                         new Class[]{extensionType}, handler));
     }
+
+    private static Object invoke(Method method, Object target, Object[] args) {
+        try {
+            if (Proxy.isProxyClass(target.getClass())) {
+                return Proxy.getInvocationHandler(target).invoke(target, method, args);
+            }
+            return MethodHandles.lookup().unreflect(method).bindTo(target).invokeWithArguments(args);
+        } catch (RuntimeException | Error e) {
+            throw e;
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
+        }
```

Comment:
`InvocationHandler.invoke()` throws `Throwable`. Seems like we could just slap a `throws Throwable` on this method and let it bubble up.

Reply:
Nope, we use this method in a lambda above, which can't throw `Throwable`.

Reply:
checked exceptions are the worst